src/audio/tdfb/tdfb_direction.c

```diff
@@ -285,9 +284,10 @@ static void level_update(struct tdfb_comp_data *cd, int frames, int ch_count, in
 	/* Calculate mean square level */
 	for (n = 0; n < frames; n++) {
 		s = *p;
+		tmp += ((int64_t)s * s);
```

Comment:
`s` is 16 bits, so casting it to 32 bits should be enough and should make this calculation faster. And yes, the external parentheses are redundant. And it actually doesn't matter where this calculation is made - here or where it was originally, right? So in its present form this change isn't improving anything